packages/esm-appointments-app/src/createDashboardLink.tsx

```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { BrowserRouter } from 'react-router-dom';
+import { DashboardExtension, type DashboardExtensionProps, type IconId } from '@openmrs/esm-framework';
+
+export const createDashboardLink = (config: Omit<DashboardExtensionProps, 'icon'> & { icon?: IconId }) => () => (
```

Comment:
We're keeping this local implementation instead of using the framework's `createDashboardLink` from `@openmrs/esm-patient-common-lib` because they have different signatures. The framework version expects `basePath` as a component prop `({ basePath }) => ...`, but our home dashboard extensions don't receive it that way.
Our version takes `basePath` in the config itself, which matches how these extensions are currently registered in `routes.json`.

It'd be great if we could have a single implementation in the framework that handles all the possible scenarios flexibly.